xds/internal/client/client_requests_counter.go

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package client
+
+import (
+	"fmt"
+	"sync"
+)
+
+func init() {
+	src.services = make(map[string]serviceInfo)
+}
+
+type servicesRequestsCounter struct {
+	mu       sync.Mutex
+	services map[string]serviceInfo
```

Comment:
A global `map[string]counter` would work. Each dropPicker would only need the `counter`, not the map.

Reply:
Made this change, now the dropPicker gets the counter.